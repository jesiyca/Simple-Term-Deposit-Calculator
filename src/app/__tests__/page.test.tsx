import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import Page from '../page'
 
describe("Home page", () => {
  it('should match snapshot', () => {
    const { container } = render(<Page />);

    expect(container).toMatchSnapshot();
  }); 
  it('should render calculator defaults successfully', () => {
    render(<Page />);

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('atMaturity-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("10330");
  }); 

  it('should successfully change deposit amount', () => {
    render(<Page />);

    fireEvent.change(screen.getByTestId('depositAmount-input'), { target: { value: "5000"}});

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("5000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('atMaturity-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("5165");
  }); 

  it('should successfully change deposit amount under minimum', () => {
    render(<Page />);

    fireEvent.change(screen.getByTestId('depositAmount-input'), { target: { value: "2"}});
    fireEvent.blur(screen.getByTestId('depositAmount-input'));

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("1000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('atMaturity-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("1033");
  }); 

  it('should successfully change deposit amount over maximum', () => {
    render(<Page />);

    fireEvent.change(screen.getByTestId('depositAmount-input'), { target: { value: "99999999"}});
    fireEvent.blur(screen.getByTestId('depositAmount-input'));

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("1500000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('atMaturity-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("1549500");
  }); 

  it('should successfully change interest rate amount', () => {
    render(<Page />);

    fireEvent.change(screen.getByTestId('interestRate-input'), { target: { value: "7.5"}});

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("7.5");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('atMaturity-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("12250");
  }); 

  it('should successfully change interest rate amount over maximum', () => {
    render(<Page />);

    fireEvent.change(screen.getByTestId('interestRate-input'), { target: { value: "30"}});
    fireEvent.blur(screen.getByTestId('interestRate-input'));

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("15");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('atMaturity-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("14500");
  }); 

  it('should successfully change interest rate amount under minimum', () => {
    render(<Page />);

    fireEvent.change(screen.getByTestId('interestRate-input'), { target: { value: "-0.1"}});
    fireEvent.blur(screen.getByTestId('interestRate-input'));

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("0");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('atMaturity-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("10000");
  }); 

  it('should successfully change investment term', () => {
    render(<Page />);

    fireEvent.change(screen.getByTestId('investmentTerm-input'), { target: { value: "12"}});

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("12");
    expect(screen.getByTestId('atMaturity-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("10110");
  }); 

  it('should successfully change investment term over maximum', () => {
    render(<Page />);

    fireEvent.change(screen.getByTestId('investmentTerm-input'), { target: { value: "100"}});
    fireEvent.blur(screen.getByTestId('investmentTerm-input'));

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("60");
    expect(screen.getByTestId('atMaturity-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("10550");
  }); 

  it('should successfully change investment term under minimum', () => {
    render(<Page />);

    fireEvent.change(screen.getByTestId('investmentTerm-input'), { target: { value: "1"}});
    fireEvent.blur(screen.getByTestId('investmentTerm-input'));

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("3");
    expect(screen.getByTestId('atMaturity-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("10028");
  }); 

  it('should successfully select monthly radio item', () => {
    render(<Page />);

    fireEvent.click(screen.getByTestId('monthly-radio'));

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('monthly-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("10335");
  }); 

  it('should successfully select quarterly radio item', () => {
    render(<Page />);

    fireEvent.click(screen.getByTestId('quarterly-radio'));

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('quarterly-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("10335");
  }); 

  it('should successfully select annually radio item', () => {
    render(<Page />);

    fireEvent.click(screen.getByTestId('annually-radio'));

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('annually-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("10334");
  });

  it('should successfully show deposit table', () => {
    render(<Page />);

    fireEvent.click(screen.getByTestId('monthly-radio'));

    expect(screen.getByTestId('depositAmount-input')).toHaveDisplayValue("10000");
    expect(screen.getByTestId('interestRate-input')).toHaveDisplayValue("1.1");
    expect(screen.getByTestId('investmentTerm-input')).toHaveDisplayValue("36");
    expect(screen.getByTestId('monthly-radio')).toBeChecked();
    expect(screen.getByTestId('finalBalance-input')).toHaveDisplayValue("10335");
    expect(screen.getByTestId('depositTable')).toBeInTheDocument();
  });
});