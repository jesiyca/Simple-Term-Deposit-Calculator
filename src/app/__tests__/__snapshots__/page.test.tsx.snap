// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should match snapshot 1`] = `
<div>
  <div
    class="grid grid-rows-[20px_1fr_20px] items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]"
  >
    <main
      class="flex flex-col gap-4 row-start-2 items-center sm:items-start"
    >
      <header>
        Simple Term Deposit Calculator
      </header>
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        data-testid="depositAmount-label"
      >
        Start deposit amount ($)
      </label>
      <input
        class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
        data-testid="depositAmount-input"
        type="number"
        value="10000"
      />
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        data-testid="interestRate-label"
      >
        Interest rate (% p.a)
      </label>
      <input
        class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
        data-testid="interestRate-input"
        type="number"
        value="1.1"
      />
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        data-testid="investmentTerm-label"
      >
        Investment term (months)
      </label>
      <input
        class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
        data-testid="investmentTerm-input"
        type="number"
        value="36"
      />
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        data-testid="interestPaid-label"
      >
        Interest paid
      </label>
      <div
        aria-required="false"
        class="grid gap-2"
        data-testid="interestPaid-radioGroup"
        dir="ltr"
        role="radiogroup"
        style="outline: none;"
        tabindex="0"
      >
        <div
          class="flex items-center space-x-2"
        >
          <button
            aria-checked="false"
            class="aspect-square h-4 w-4 rounded-full border border-primary text-primary shadow focus:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
            data-radix-collection-item=""
            data-state="unchecked"
            data-testid="monthly-radio"
            id="r1"
            role="radio"
            tabindex="-1"
            type="button"
            value="monthly"
          />
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            data-testid="monthly-label"
            for="r1"
          >
            Monthly
          </label>
        </div>
        <div
          class="flex items-center space-x-2"
        >
          <button
            aria-checked="false"
            class="aspect-square h-4 w-4 rounded-full border border-primary text-primary shadow focus:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
            data-radix-collection-item=""
            data-state="unchecked"
            data-testid="quarterly-radio"
            id="r2"
            role="radio"
            tabindex="-1"
            type="button"
            value="quarterly"
          />
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            data-testid="quarterly-label"
            for="r2"
          >
            Quarterly
          </label>
        </div>
        <div
          class="flex items-center space-x-2"
        >
          <button
            aria-checked="false"
            class="aspect-square h-4 w-4 rounded-full border border-primary text-primary shadow focus:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
            data-radix-collection-item=""
            data-state="unchecked"
            data-testid="annually-radio"
            id="r3"
            role="radio"
            tabindex="-1"
            type="button"
            value="annually"
          />
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            data-testid="annually-label"
            for="r3"
          >
            Annually
          </label>
        </div>
        <div
          class="flex items-center space-x-2"
        >
          <button
            aria-checked="true"
            class="aspect-square h-4 w-4 rounded-full border border-primary text-primary shadow focus:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
            data-radix-collection-item=""
            data-state="checked"
            data-testid="atMaturity-radio"
            id="r4"
            role="radio"
            tabindex="-1"
            type="button"
            value="atMaturity"
          >
            <span
              class="flex items-center justify-center"
              data-state="checked"
            >
              <svg
                class="h-3.5 w-3.5 fill-primary"
                fill="none"
                height="15"
                viewBox="0 0 15 15"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.4669 3.72684C11.7558 3.91574 11.8369 4.30308 11.648 4.59198L7.39799 11.092C7.29783 11.2452 7.13556 11.3467 6.95402 11.3699C6.77247 11.3931 6.58989 11.3355 6.45446 11.2124L3.70446 8.71241C3.44905 8.48022 3.43023 8.08494 3.66242 7.82953C3.89461 7.57412 4.28989 7.55529 4.5453 7.78749L6.75292 9.79441L10.6018 3.90792C10.7907 3.61902 11.178 3.53795 11.4669 3.72684Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            data-testid="atMaturity-label"
            for="r4"
          >
            At Maturity
          </label>
        </div>
      </div>
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        data-testid="finalBalance-label"
      >
        Final balance ($)
      </label>
      <input
        class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
        data-testid="finalBalance-input"
        disabled=""
        type="number"
        value="10330"
      />
    </main>
  </div>
</div>
`;
